filebeat: fix Redis auth method to support username/password (#44137)
elastic/beats: CHANGELOG.next.asciidoc

`log.syslog.appname`. The field `container.partial` is replaced by the tag `partial_message` if it was `true`,
otherwise no tag is added. {issue}42208[42208] {pull}42403[42403]
- Fixed race conditions in the global ratelimit processor that could drop events or apply rate limiting incorrectly.

> - Fixed Redis input in Filebeat to support username and password authentication. {pull}44137[44137]

Reviewer: From the diff it looks like we had already supported the password authentication. It's also documented https://www.elastic.co/docs/reference/beats/filebeat/filebeat-input-redis#redis-password

So, what changed here functionally?

I think we just pass it differently to the client now, is that correct? Is it just refactoring then?

Author: I encountered an auth failure when using Filebeat with Redis ACL (username + password).
Even though both fields were set, Filebeat failed to connect.
This was because the old code passed only the username via `DialOption`, and manually called `AUTH <password>`, which doesn't work for ACL users.
Redis requires `AUTH <username> <password>` in that case.
This PR fixes the issue by passing both `username` and `password` via `DialOption`, which redigo handles correctly.
Metricbeat already follows this pattern:
[https://github.com/elastic/beats/blob/6e3634f514fad19797d34d4f5a11fda4af45e2cf/metricbeat/module/redis/redis.go#L211](https://github.com/elastic/beats/blob/6e3634f514fad19797d34d4f5a11fda4af45e2cf/metricbeat/module/redis/redis.go#L211)

Reviewer: Then I'd suggest to word the changelog entry differently:

```suggestion
- Fixed password authentication for ACL users in the Redis input of Filebeat. {pull}44137[44137]
```

Author: Updated changelog wording as suggested, thanks!
